### DISCUSSION TITLE
Question about vici.Session.listen API

### DISCUSSION THREAD

**Discussion by sharathbhatp** [strongswan/strongswan#2581] [General]
Hi

Can running a separate process that listens to IKE events like "ike-updown", "child-updown", "ike-rekey" using vici.Session.listen() can stress charon process when there can be 1000+ sessions? Are there any known limitations?

Thanks

**Comment by tobiasbrunner** [comment]
There will be some impact as a vici message has to be prepared for each event that has subscribers, but the actual events are sent asynchronously, so it might not be that significant.

**Reply by sharathbhatp** [reply]
Thanks. What will be the timestamp on the event? Is it the time when the event occured or the time when the event is dispatched to vici asyncronously?

**Reply by tobiasbrunner** [reply]
What do you mean with "timestamp"?

**Reply by sharathbhatp** [reply]
The "time of event" field in the message sent to vici client

**Reply by tobiasbrunner** [reply]
There is no such field.
